### PR TITLE
Washington, Idaho, and the statistics cache

### PR DESCRIPTION
**Pull request by ryanrolds**
Requires: https://github.com/Hack4Eugene/speedupamerica-migrator/pull/8

This PR updates the nightly data processing to load Washington, Idaho, and Oregon. It also adds a step to the nightly processing that pre-computes statistics for Zips, Census Tracts, and ISPs. Also, the results pages are updated to use the cached data to quickly serve pages. Any filter selections that are no pre-computed use the original method of calculating data in at the time of the request.

To get the initial page load time fast I had to sacrifice some filter options and the census tract map at the bottom. We can debate the value of these features and if they are worth the overhead, I don't believe that they are.

When this PR is merged I will have to wipe all M-Lab submissions from Test and run the data processing. The same is true for then it's deployed to Production, which I make sure to do in the middle of the night.

> The submissions model changes are large enough to be not automatically shown by GitHub, make sure to expand them.

### CODE REVIEW

File: app/assets/javascripts/mapboxgl.coffee [hunk 1]
```diff
@@ -39,12 +39,7 @@ window.initialize_mapboxgl = (elmID) ->
 
 get_map_loader = (map) ->
   map_id = map.getContainer().id
-
-  if map_id == 'all_results_map'
-    loader_id = '#loader'
-  else if map_id == 'zip_code_map'
-    loader_id = '#stats_loader'
-
+  loader_id = '#loader'
```
Comment (Kourchenko) [inline app/assets/javascripts/mapboxgl.coffee]: Is this in an effort to move all stats to their own page?

File: app/classes/submissions_importer.rb [hunk 2]
```diff
@@ -37,120 +37,135 @@ def self.attribute_val(row, attributes, name)
     row['f'][attributes.index(name)]['v']
   end
 
-  def self.create_submissions(data, test_type)
```
Comment (Kourchenko) [inline app/classes/submissions_importer.rb]: Are we reworking the format of our submissions? Do I understand this correctly?

File: app/classes/submissions_importer.rb [hunk 3]
```diff
@@ -37,120 +37,135 @@ def self.attribute_val(row, attributes, name)
...
-    zip_codes = "'#{zips.join("','")}'"
+    country_code = 'US'
+    regions = ['OR', 'WA', 'ID']
+    end_time = Date.today.strftime("%Y-%m-%d")
+
+    regions.each do |region|     
 
-    upload_query = upload_query(zip_codes)
-    download_query = download_query(zip_codes)
+      while true #we may need to get multiple batches
```
Comment (Kourchenko) [inline app/classes/submissions_importer.rb]: Nothing to see here... 🚶 💨

Comment (Kourchenko) [reply app/classes/submissions_importer.rb]: Does Rails do asynchronous calls, or does this go quickly such that it doesn't block the main thread?

File: app/assets/javascripts/mapboxgl.coffee (hunk 1 above)
Comment (ryanrolds) [reply app/assets/javascripts/mapboxgl.coffee]: No, there just isn't a 2nd map and the branch above wasn't needed anymore.

File: app/classes/submissions_importer.rb (hunk 2 above)
Comment (ryanrolds) [reply app/classes/submissions_importer.rb]: I moved the function lower in the file, yes.

File: app/classes/submissions_importer.rb (hunk 3 above)
Comment (ryanrolds) [reply app/classes/submissions_importer.rb]: The Submission Import runs in its own container/runtime. So, it doesn't really block anything as that container is only handling data import/processing.

**Comment by antonioortegajr** [comment]
I agree the removal of filter options and the census tract map do not seem essential at this time to me. Especially when regional coverage is a higher goal over additional maps or filters.

**Pull request merged**